tests/bud.bats

```diff
@@ -25,6 +25,60 @@ load helpers
   expect_output --substring "options use-vc"
 }
 
+# Test for https://github.com/containers/buildah/pull/4295
+@test "build test warning for preconfigured TARGETARCH, TARGETOS, TARGETPLATFORM or TARGETVARIANT" {
+  mkdir -p ${TEST_SCRATCH_DIR}/bud/platform
+
+  # buildah should not warn when TARGETARCH, TARGETOS, TARGETPLATFORM or TARGETVARIANT is indirectly set by cli options
+  cat > ${TEST_SCRATCH_DIR}/bud/platform/Dockerfile << _EOF
+FROM alpine
+ARG TARGETARCH
+COPY . .
+_EOF
+  run_buildah build $WITH_POLICY_JSON --platform linux/amd64 -t source -f ${TEST_SCRATCH_DIR}/bud/platform/Dockerfile
+  assert "$output" !~ "--build-arg TARGETARCH"
```

Comment:
Too explicit. It's much better to have a looser test here, such that any warning will cause a test failure.